### PR TITLE
Fixed #30990 -- Fixed example output in 'z' date format docs.

### PR DESCRIPTION
**Pull request by bmispelon**

### CODE REVIEW

File: tests/utils_tests/test_dateformat.py [hunk 1]
```diff
@@ -115,6 +115,10 @@ def test_dateformat(self):
     def test_day_of_year_leap(self):
         self.assertEqual(dateformat.format(datetime(2000, 12, 31), 'z'), '366')
 
+    def test_iso_year_edgecase(self):
+        d = datetime(2003, 12, 29)
+        self.assertEqual(dateformat.format(d, 'o'), '2004')
```
Comment (felixxm) [inline tests/utils_tests/test_dateformat.py]: This is already tested in `template_tests.filter_tests.test_date.DateTests.test_date04()/test_date05()`. I will remove this test.

Comment (bmispelon) [reply tests/utils_tests/test_dateformat.py]: Oh, you're right! I didn't think to check for a test in a different folder.

**Comment by felixxm** [comment]
@bmispelon Thanks :+1: I pushed minor edits and reorganized commits.

**Comment by bmispelon** [comment]
Excellent, thanks! :sparkles:

**Pull request merged**